cpp/include/cudf/detail/contiguous_split.hpp

```diff
@@ -67,7 +67,7 @@ class metadata_builder {
    * @brief Destructor that will be implemented as default, required because metadata_builder_impl
    * is incomplete at this stage.
    */
-  ~metadata_builder() = default;
+  ~metadata_builder();
```

Comment:
if this destructor is defined in the header like before, I get compilation errors in this new PR. I just introduced this class in  a prior PR https://github.com/rapidsai/cudf/pull/13232, and actually changed where this was defined due to a comment. I am happy to dig into it more if people think it's odd.

Reply:
That is because in the destructor, the content of `std::unique_ptr<detail::contiguous_split_state> state;` is destroyed. There is no definition of `detail::contiguous_split_state` in this header thus you can't compile it.

Reply:
Previously, this issue didn't show up because `metadata_builder` was instantiated only after defining `metadata_builder_impl` in `pack.cpp`. Now `metadata_builder` is instantiated in `contiguous_split.cu` which doesn't have a definition for `metadata_builder_impl`, triggering the compiler issue.

Reply:
Thanks for the feedback @ttnghia